xtask/codegen/src/generate_syntax_kinds.rs

```diff
@@ -200,8 +200,8 @@ pub fn generate_syntax_kinds(grammar: KindsSrc, language_kind: LanguageKind) ->
     };
 
     let ast = quote! {
-        #![allow(clippy::all)]
-        #![allow(bad_style, missing_docs, unreachable_pub)]
+        #![expect(clippy::all)]
+        #![expect(bad_style)]
```

Comment:
This is the kind of thing that I would actually prefer keeping our existing `allow`s instead. Since the `expect` is inside generated code, these diagnostics will bubble up create extra friction when setting up new languages (which is already quite high, and we're still planning to add more languages). Not ideal for DX.